### PR TITLE
[CI] Update scripts/tests/run_test_suite.py so it can run darwinframeworktool.py

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem

Now that `Darwin-framework-tool` is compatible with YAML tests, it needs to be able to be used by `scripts/tests/run_test_suite.py`.

This is an other step forward to get rid of the codegen version of `darwin-framework-tool` tests.

### CODE REVIEW

File: examples/darwin-framework-tool/commands/common/CHIPCommandBridge.h [hunk 1]
```diff
@@ -45,6 +45,8 @@ class CHIPCommandBridge : public Command {
         AddArgument("paa-trust-store-path", &mPaaTrustStorePath,
             "Path to directory holding PAA certificate information.  Can be absolute or relative to the current working "
             "directory.");
+        AddArgument("storage-directory", &mStorageDirectory,
+            "Directory to place chip-tool's storage files in.  Defaults to $TMPDIR, with fallback to /tmp");
```
Comment (bzbarsky-apple) [inline examples/darwin-framework-tool/commands/common/CHIPCommandBridge.h:49]:
```suggestion
            "Directory to place darwin-framework-tool's storage files in.  Defaults to $TMPDIR, with fallback to /tmp");
```
But is this really true?  If we are only doing this for interactive commands, I fully expect this thing is pretty much ignored.....

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/darwin-framework-tool/commands/common/CHIPCommandBridge.h (hunk 1 above)
Comment (vivien-apple) [reply examples/darwin-framework-tool/commands/common/CHIPCommandBridge.h]: You are correct that this is not really true at the moment. The current storage mechanism use some built-in preference system. I wonder if I should:
 - add a custom persistent storage backend that stores things in files. But I think that was an issue to run GitHub workflows and that is why this thing as been implemented using preferences. @krypton36  ?
 - update the mechanism to works with a domain for `darwin-framework-tool`. For example `--storage-directory` may change the domain used to save preferences (from `com.apple.chiptool` to `com.apple.chiptool-{storage-directory}`)

Comment (vivien-apple) [reply examples/darwin-framework-tool/commands/common/CHIPCommandBridge.h]: In any case this should probably be a followup unless this is causing issues. I would like to move forward and try to get https://github.com/vivien-apple/connectedhomeip-1/tree/MatterYamlTests_DarwinFrameworkTool_All_Tests in.

The mentioned branch gets the command `/scripts/tests/run_test_suite.py --runner chip_tool_python --chip-tool out/debug/standalone/darwin-framework-tool  run --test-timeout-seconds 120` to works successfully and it depends on this one ;)

Comment (bzbarsky-apple) [reply examples/darwin-framework-tool/commands/common/CHIPCommandBridge.h]: I think for now accurately documenting the option as "it's here for API compat with chip-tool but does nothing" is the thing to do.

File: scripts/tests/run_test_suite.py [hunk 2]
```diff
@@ -160,7 +160,7 @@ def main(context, dry_run, log_level, target, target_glob, target_skip_glob,
     # Figures out selected test that match the given name(s)
     if runtime == TestRunTime.CHIP_REPL_PYTHON:
         all_tests = [test for test in chiptest.AllReplYamlTests()]
-    elif runtime == TestRunTime.CHIP_TOOL_PYTHON:
+    elif runtime == TestRunTime.CHIP_TOOL_PYTHON and os.path.basename(chip_tool) != "darwin-framework-tool":
```
Comment (andy31415) [inline scripts/tests/run_test_suite.py:163]: This feels unclean. How about another constant of DARWIN_TOOL_PYTHON?

Comment (andy31415) [reply scripts/tests/run_test_suite.py:163]: Unsure what the blast radius of that is... I checked how often CHIP_TOOL_PYTHON is used and it does not seem to be used that much (just 3 times) and I imagine some logic needed for `chip_tool_with_python`.

If that is really impossible then maybe just some override on what tests to run could be used. the extra basename check and fallback into "all chip tool tests" seems not ideal.

Comment (vivien-apple) [reply scripts/tests/run_test_suite.py:163]: I have some other patches that i would push once this one lands. Everything is in https://github.com/vivien-apple/connectedhomeip-1/tree/MattterYamlTests_RemoveCodegenTests and is mostly ready to be made as PRs.
In https://github.com/vivien-apple/connectedhomeip-1/commit/e747328dffd87a636bcc263551c4267ab803eac3 I am updating the TestRuntime enum to be cleaner! 
Would you be fine with a followup ?

Comment (andy31415) [reply scripts/tests/run_test_suite.py:163]: Ok, I take that this proves that this is indeed getting better in time, am happy with that.

**Review by andy31415 — APPROVED** [review]

**Pull request merged**